unit_tests/stubs/apt.py

```diff
@@ -14,3 +14,12 @@ def queue_install(self, packages):
 
     def add_source(self, source, key=None):
         self.sources.append((source, key))
+
+    def get_package_version(self, package, full_version=False):
+        return '2.150.1'
+
+
+class AptStubLegacyJenkinsVersion(AptStub):
+    """Testable stub that returns an older version of Jenkins"""
+    def get_package_version(self, package, full_version=False):
+        return '2.128.1'
```

Comment:
Instead of adding a whole new subclass, it'd be a bit nicer to just add a new attribute (e.g. self._package_version) that tests can modify with a method like ```self.apt.set_legacy_jenkins_version()```. The rational is that composition is generally preferable over inheritance, because the latter can easily lead to combinatorial explosion.

This just a nit/hint, not blocking, I'm fine if you prefer to keep things as they are.